cmd/clusterdeployment/listresources.go

```diff
@@ -0,0 +1,193 @@
+package clusterdeployment
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	gcpv1alpha1 "github.com/openshift/gcp-project-operator/pkg/apis/gcp/v1alpha1"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/spf13/cobra"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/osdctl/pkg/k8s"
+	"github.com/openshift/osdctl/pkg/printer"
+)
+
+// newCmdList implements the list command to list
+func newCmdListResources(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newListResourcesOptions(streams, flags)
+	lrCmd := &cobra.Command{
+		Use:               "listresources",
+		Short:             "List all resources on a hive cluster related to a given cluster",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(runListResources(cmd, ops))
+		},
+	}
+	lrCmd.Flags().StringVarP(&ops.clusterId, "cluster-id", "C", "", "Cluster ID")
+
+	return lrCmd
+}
+
+// listOptions defines the struct for running list command
+type listResourcesOptions struct {
+	flags *genericclioptions.ConfigFlags
+	genericclioptions.IOStreams
+	kubeCli   client.Client
+	clusterId string
+}
+
+func newListResourcesOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *listResourcesOptions {
+	return &listResourcesOptions{
+		flags:     flags,
+		IOStreams: streams,
+	}
+}
+
+func (o *listResourcesOptions) complete(_ *cobra.Command, _ []string) error {
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runListResources(cmd *cobra.Command, o *listResourcesOptions) error {
+	if o.clusterId == "" {
+		return cmdutil.UsageErrorf(cmd, "No cluster ID specified, use -C to set one")
```

Comment:
I added support to use clusterid, external and cluster name.. maybe to phase 2 add support to any value?

Reply:
is there a utility function I can use to extract the cluster ID from those identifiers?